src/main/kotlin/lotto/domain/LottoNumbers.kt

```diff
@@ -0,0 +1,53 @@
+package lotto.domain
+
+import lotto.argumentError
+
+class LottoNumbers(
+    private val numbers: List<LottoNumber>
+) {
+    init {
+        check(numbers.size == LOTTO_NUMBER_SET) {
+            argumentError("로또는 중복 없는 6자리 번호여야합니다.")
+        }
+    }
+
+    fun matches(inputNumbers: LottoNumbers): Int {
+        return numbers.filter { number ->
+            inputNumbers.numbers.contains(number)
+        }.size
+    }
```

Comment:
```suggestion
    fun matches(inputNumbers: LottoNumbers): Int {
         return numbers.count{
			number -> inputNumbers.numbers.contains(number)
	    }
    }
```

count를 이용해도 괜찮을 것 같습니다 :)